tests/extensions/test_extensions.py

```diff
@@ -130,3 +130,19 @@ def do(self, *args):
     comp.do.assert_called_once_with('before_training')
     a.do.assert_called_once_with('after_training')
     b.do.assert_called_once_with('after_batch')
+
+
+def test_simple_extension_before_batch_callback():
+
+    class Foo(SimpleExtension):
+        def __init__(self, **kwargs):
+            self.do = Mock()
+            super(Foo, self).__init__(**kwargs)
+
+        def do(self, which_callback, *args):
```

Comment:
You're right. I actually copied that part from another test and it seems there are a couple of locations in this file where this happens. Again, also for future refactoring :-).